### ISSUE TITLE
allOf fails to resolve RELATIVE ref

### ISSUE THREAD

**Issue by EtonDoze** [swagger-api/swagger-parser#616]
Here is an example
```
.
├── models
│   ├── pet.json
│   └── fancy_pet.json    #inherits pet.json using allOf
└── parent.json
```
### parent.json
```
{
  "swagger": "2.0",
  "paths": {
    "/pet": {
      "get": {
        "responses": {
          "200": {
            "schema": {
              "$ref": "./models/pet.json"
            }
          }
        }
      }
    },
    "/fancy-pet": {
        "responses": {
          "200": {
            "schema": {
              "$ref": "./models/fancy_pet.json"
            }
          }
        }
      }
    }
  }
}
```

### models/`pet.json`
```
{
  "type": "object",
  "properties": {
    "id": {
      "type": "integer",
      "format": "int64"
    },
    "name": {
      "type": "string",
      "example": "Cooper"
    }
  }
}
```

### models/`fancy_pet.json`
Notice here that `pet.json` is referenced using relative path which is in the same `models` folder.
```
{
  "allOf": [
    {
      "$ref": "./pet.json"
    },
    {
      "properties": {
        "breed": {
          "type": "string",
          "example": "Labrador"
        }
      }
    }
  ]
}
```

Issue: parser fails with error `java.lang.RuntimeException: Unable to load RELATIVE ref: ./pet.json`. This happens because the `pet.json` is resolved RELATIVE to `parent.json` instead of `fancy_pet.json`. I expect parser needs to resolve `pet.json` w.r.t. to the origin i.e. `fancy_pet.json`

I have already found a fix for this issue and PR is on its way.

CC: @tzimisce012

**Comment by gracekarina** [comment]
@EtonDoze Hi, is this related to https://github.com/swagger-api/swagger-parser/issues/567?

**Comment by EtonDoze** [comment]
@gracekarina , its similar #604 but not same.
I created a PR #617 to fix this #604.

**Comment by gracekarina** [comment]
@EtonDoze please check out the merge, to close the ticket

**Comment by EtonDoze** [comment]
Closing it as its fixed via #617
